packages/core/src/odata/v4/payload-value-converter.ts

```diff
@@ -107,9 +107,22 @@ function durationToEdmDuration(value: moment.Duration): string {
 }
 
 function timeToEdmTimeOfDay(value: Time): string {
-  return `${value.hours}:${value.minutes}:${value.seconds}`;
+  const hoursPadded = value.hours.toString(10).padStart(2, '0');
+  const minutesPadded = value.minutes.toString(10).padStart(2, '0');
+  const secondsPadded = padSeconds(value.seconds);
+  return `${hoursPadded}:${minutesPadded}:${secondsPadded}`;
```

Comment:
I think you can repeat yourself a little less, also see suggestion below.
```suggestion
  return [value.hours, value.minutes, value.seconds].map(timeComponent => padTimeComponent(timeComponent)).join(':');
```